scenario/uploaddata.go

```diff
@@ -31,8 +35,98 @@ type (
 	}
 )
 
-const datafileEndpoint = "api/v1/qix-datafiles"
-const refererPath = "%s/explore/spaces/%s/data"
+const (
+	datafileEndpoint = "api/v1/qix-datafiles"
+)
+
+type (
+	TempFile struct {
+		ID  string
+		URL string
+	}
+
+	TempFileClient struct {
+		tusClient  *tus.Client
+		maxRetries int
+	}
+)
+
+func NewTempFileTUSClient(sessionState *session.State, connection *connection.ConnectionSettings, chunkSize int64, maxRetries int) (*TempFileClient, error) {
```

Comment:
Maybe better to put in it's own files, in a helpers package or own package. Especially since it's used by more than one action.

Reply:
otherwise lgtm, have not verified traffic though

Reply:
Yes, I'll put it in a new package.